### DISCUSSION TITLE
new to GitHub

### DISCUSSION THREAD

**Discussion by Wilsori** [community/community#63884] [Programming Help] [answered]
I follow exactly in the youtube video how to start with Raspberry Pi Pico and when run appears "machine" not found
What can I do?

**Comment by oftharsh** [comment]
The error message "machine not found" typically indicates that the `machine` module, which is part of the MicroPython standard library, is not available or not imported correctly in your Python script on the Raspberry Pi Pico.

Here are a few steps you can take to troubleshoot and resolve this issue:

1. **Check MicroPython Installation**: Make sure you have MicroPython installed on your Raspberry Pi Pico. If you haven't flashed MicroPython onto the Pico yet, you'll need to follow the instructions to install it.

2. **Import Correctly**: In your Python script, make sure you are importing the `machine` module correctly. The import statement should look like this:

   ```python
   import machine
   ```

3. **Check Script Location**: Ensure that your Python script is located in the same directory as the `machine` module. If you are using an integrated development environment (IDE) or code editor, the script's working directory might be different from where the `machine` module is located.

4. **Verify File Name**: Check that your script's file name does not conflict with any reserved names, such as `machine.py`. This could lead to conflicts when trying to import the `machine` module.

5. **Reflash MicroPython**: If you continue to encounter issues, you might consider reflashing MicroPython onto the Raspberry Pi Pico. This can help ensure that all required modules are properly installed.

6. **Check MicroPython Version**: Make sure you are using a version of MicroPython that includes the `machine` module. The `machine` module is typically included in most MicroPython distributions, but it's worth confirming.

7. **IDE or Terminal**: Check if you are running the Python script from an integrated development environment (IDE) or directly from a terminal. Some IDEs might have their own configurations that affect module availability.

8. **Double-Check Syntax**: Make sure there are no syntax errors or typos in your code that might be causing the issue.

If you've gone through these steps and are still encountering the "machine not found" error, please provide more details about how you are running the script, where the script is located, and any additional error messages you are seeing. This will help in diagnosing the issue more accurately.

**Comment by queenofcorgis** [comment, marked as answer]
Thanks for posting in the GitHub Community, @Wilsori ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by shivvamm** [comment]
It looks like you're encountering an issue related to the "machine" module not being found when working with a Raspberry Pi Pico. This error typically indicates that the required Python module is missing or not properly installed. Here's what you can do to troubleshoot and resolve this issue:

Check Your Code: Double-check your Python code to ensure that you have correctly imported the "machine" module. It should look something like this:

```python
import machine
```
Check Python Version: Make sure you're using a compatible version of Python. The "machine" module is typically associated with MicroPython, which is a lightweight implementation of Python for microcontrollers. Ensure you're using MicroPython and that it's properly installed on your Raspberry Pi Pico.

Install MicroPython: If you haven't installed MicroPython on your Raspberry Pi Pico, you'll need to do so. MicroPython provides the "machine" module for hardware-related operations. Follow the official documentation or guides to install MicroPython on your Pico.

Verify File Naming: Ensure that the file name for your Python script ends with the ".py" extension. For example, if your script is named "my_script.py", make sure you're running it as python my_script.py.

Upload Files Properly: If you're transferring your Python script to the Raspberry Pi Pico, ensure that you're uploading it correctly. Sometimes, issues can arise if the file transfer isn't successful.

Update MicroPython: If you already have MicroPython installed, check if there's an updated version available and consider updating it to the latest version.

Check Virtual Environment: If you're working within a virtual environment, make sure that the environment is activated before running your script. The "machine" module needs to be available in the active environment.

Reflash Firmware: If you're still having trouble, you might consider reflashing the MicroPython firmware onto your Raspberry Pi Pico. This can help ensure that all the necessary modules are present.

Community Support: If none of the above steps work, consider reaching out to the Raspberry Pi Pico community forums, MicroPython forums, or the source of the YouTube video you followed. Others may have encountered similar issues and might be able to provide specific guidance.

Remember that the exact steps and solutions might vary depending on the specific details of your setup and the version of MicroPython you're using. Make sure to carefully follow the installation and setup instructions provided by the MicroPython documentation or the YouTube video you're following.

**Reply by Wilsori** [reply]
Dear shivvamm: thank so much for your reply. It is hard to understand and to check your recomendations. I am new with Raspeberry pi Pico. I was working with PIC microchip 18f2550 and I foundthe necessity to work with ADC 12 bits and Amzon suggested the Raspeberry.. imaginig was the ADC only.. but I found dificulties. I follow the indications from YouTube and load something into the Raspberry but when I started to excute the samples found machine was missing. After loading I understanded to need anIDE or teminal and Thjommy was suggested... again following what was in the guide.. 
You asked me if I have load the Microphyton?... I do not know what is really loeaded..
Python 3.10.11 (C:\Users\wilso\AppData\Local\Programs\Thonny\python.exe)
>>> 
That seems what is loaded
I have no idea how can I make all those checks you point out..
Maybe would be necessary to erase everything and start again?
I wrote
import machine

and reply Python3.10.11... 
I would appreciate your support with more detail as a new player
thank you in advance

**Reply by shivvamm** [reply]
Hey @Wilsori I'm glad to help you didn't intend to provide misleading solution that's why I put a brief wish it helped you in anyway

**Comment by 0x7n6** [comment]
Ensure that you have the necessary dependencies installed. The "machine" module is part of the MicroPython library, which is the default firmware for Raspberry Pi Pico. Make sure you have MicroPython installed on your Pico board.

Verify that you have the correct firmware flashed on your Raspberry Pi Pico. If you don't have the MicroPython firmware installed, you can follow the official MicroPython documentation to flash the firmware onto your Pico.

If you have already flashed the MicroPython firmware and the error persists, it's possible that the "machine" module is missing or corrupted. In this case, you can try reinstalling the MicroPython firmware on your Raspberry Pi Pico. Download the latest version of the firmware from the official MicroPython website and follow the flashing instructions.

Double-check your code and make sure that you import the "machine" module correctly. The import statement should be import machine. Ensure that there are no typos or syntax errors in your code.

If you're using an integrated development environment (IDE) or a code editor, ensure that it is properly configured to work with Raspberry Pi Pico. Check the documentation or settings of your IDE to make sure it is using the correct Python environment and is able to communicate with the Pico board.

**Reply by Wilsori** [reply]
Thank you for your patience support.
I found the module machine.py :
C:\Users\wilso\AppData\Local\Programs\Thonny\Lib\site-packages\thonny\plugins\microbit\api_stubs
But when was loaded appeared this message:

WARNING: Your `C:\Users\wilso\AppData\Local\Programs\Thonny\Lib\site-packages\thonny\plugins\microbit\api_stubs\collections.py` can shadow the library module 'collections'. Consider renaming or moving it!

>>> 
I do not know what that is meaning...
How can I check if the MicroPhyton has been installed into my Raspeberry pi Pico?
and which version?

Thank you in advance
